prov/sockets/src/sock_ep.c

```diff
@@ -1617,8 +1617,10 @@ int sock_ep_get_conn(struct sock_ep *ep, struct sock_tx_ctx *tx_ctx,
 	if (conn == SOCK_CM_CONN_IN_PROGRESS)
 		conn = sock_ep_connect(ep, av_index);
 
-	if (!conn)
-		return -errno;
+	if (!conn) {
+		SOCK_LOG_ERROR("Error in connecting: %d\n", -errno);
+		return -FI_EAGAIN;
+	}
```

Comment:
@shefty - Thanks. Added check for EINPROGRESS, which will avoid the case where it can ignore real errors.